SCPDiscordPlugin/LogicRole.cs

```diff
@@ -0,0 +1,145 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace SCPDiscord
+{
+	public enum LogicType
+	{
+		None,
+		And,
+		Or,
+		NotAnd,
+		PrimaryOnly,
+		Xor,
+		NotOr
+	}
+
+	[JsonObject]
+	public class LogicRole
+	{
+		public LogicType Type { get; set; }
+		public List<ulong> Roles { get; set; }
+		[CanBeNull] public List<string> Commands { get; set; }
+		[CanBeNull] public Dictionary<int, LogicRole> Children { get; set; }
+
+		public bool IsPermitted(List<ulong> userRoles)
+		{
+			return Type switch
+			{
+				LogicType.None => true,
```

Comment:
Oops, here should be false. Means we never go execute commands and go into childrens